hs_core/models.py

```diff
@@ -629,6 +638,30 @@ def update(cls, element_id, **kwargs):
                     raise ValidationError('Relation type:%s cannot be updated since '
                                           'isHostedBy relation already exists.' % rel.type)
 
+            # avoid changing this relation to an existing relation of same type and same value
+            metadata_obj = kwargs['content_object']
+            metadata_type = ContentType.objects.get_for_model(metadata_obj)
+            qs = Relation.objects.filter(type=kwargs['type'],
+                                         value=kwargs['value'],
+                                         object_id=metadata_obj.id,
+                                         content_type=metadata_type)
+            if rel.type == kwargs['type']:
```

Comment:
I think the following should be sufficient to avoid duplicates:

```
if qs.exists() and qs.first() != rel:
     # this update will create a duplicate relation element
     raise ValidationError('A relation element of the same type and value already exists.')
```